lib/ansible/modules/notification/irc.py

```diff
@@ -185,14 +191,15 @@ def send_msg(msg, server='localhost', port='6667', channel=None, nick_to=None, k
     if use_ssl:
         irc = ssl.wrap_socket(irc)
     irc.connect((server, int(port)))
+
     if passwd:
-        irc.send('PASS %s\r\n' % passwd)
-    irc.send('NICK %s\r\n' % nick)
-    irc.send('USER %s %s %s :ansible IRC\r\n' % (nick, nick, nick))
+        irc.send(safe('PASS %s\r\n' % passwd))
+    irc.send(safe('NICK %s\r\n' % nick))
+    irc.send(safe('USER %s %s %s :ansible IRC\r\n' % (nick, nick, nick)))
```

Comment:
From the error message that the user is getting, it looks like this needs to be bytes.  So this will still fail on python3.  These should probably all use to_bytes instead.

Reply:
that fails in python2 which requires strings, hence my use of `safe` to convert depending on version

I actually tested my changes with py2 and py3, see in #ansilbe-devel for 'notbcoca'

Reply:
My testing confirms that this needs to be bytes on both python2 and python3.  If it's text then it fails on python3.  If it's the opposite of native (text on python2 and bytes on python3) then it fails with non-ascii characters on python2.